cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/JiraBucketWidget.kt

```diff
@@ -0,0 +1,40 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.widget.AsyncWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import com.cognifide.cogboard.CogboardConstants as CC
+
+class JiraBucketWidget(vertx: Vertx, config: JsonObject) : AsyncWidget(vertx, config) {
+
+    private val bucketQueries: JsonArray = config.getJsonArray("bucketQueries")
+    private val bucketNames: JsonArray = JsonArray()
+    private val issueCounts: JsonArray = JsonArray()
+
+    override fun handleResponse(responseBody: JsonObject) {
+        val response = JsonObject()
+        val issues = responseBody.getJsonArray("issues").size()
+
+        bucketNames.add(responseBody.getString("requestId"))
+        issueCounts.add(issues)
+        response.put("bucketNames", bucketNames)
+        response.put("issueCounts", issueCounts)
+        send(JsonObject()
+                .put(CC.PROP_CONTENT, response))
+    }
+
+    override fun updateState() {
+        if (url.isNotBlank()) {
+            for (bucketQuery in bucketQueries.list) {
+                if (bucketQuery is JsonObject) {
+                    val bucketName = bucketQuery.getString("bucketName")
+                    val jqlQuery = bucketQuery.getString("jqlQuery")
+                    httpGet(url = "$url/jira/rest/api/2/search?jql=$jqlQuery", requestId = bucketName)
```

Comment:
I would rather see `requestId` set from some id generated in frontend

Reply:
Forgive my ignorance, I'm completely new to this and just looking around . What's the purpose of this `requestId` parameter? Is it meant to serve as a correlation id in log inspection?

Reply:
done

Reply:
We are calling jira asynchronously and we need to match request id to match data with correct response.